src/components/interaction-designer/block/BlockToolbar.vue

```diff
@@ -84,40 +85,32 @@ export class BlockToolbar extends mixins(Lang) {
 
   isDeleting = false
 
-  handleDeleteBlock(): void {
-    this.block_deselect({blockId: this.block.uuid})
-    this.flow_removeBlock({blockId: this.block.uuid})
+  async handleDeleteBlock(): Promise<void> {
+    await this.block_deselect({blockId: this.block.uuid})
+    await this.flow_removeBlock({blockId: this.block.uuid})
     this.isDeleting = false
+    await this.takeSnapshot()
     this.$emit('after-delete')
   }
 
-  handleDuplicateBlock(): void {
-    this.flow_duplicateBlock({blockId: this.block.uuid}).then((duplicatedBlock) => {
-      this.$router.replace({
-        name: 'block-selected-details',
-        params: {blockId: duplicatedBlock.uuid},
-      })
```

Comment:
used the chance to refactor `.then()` into `await`

Reply:
@bulatgab , I'm curious why do we need to replace then to await?
- with `then`, we only call the instruction if the main promise was a success
- with await, we always call the next instruction even the main promise was a failure (thus, duplicatedBlock might not be defined)

Reply:
To me it was about code style only. I think we are mainly using `async+await` in our codebase rather than `.then()`.

In terms of equivalence - I might be wrong but I think if the promise rejects on this line:
```
const duplicatedBlock = await this.flow_duplicateBlock({blockId: this.block.uuid})
```
it would throw an error and the rest of the code in the function would be skipped.

I guess we could add try-catch indeed (as it's often done around await) just to be on the safe side! Would that work?

Reply:
Fixed